server/functions/index.js

```diff
@@ -3,31 +3,73 @@ const functions = require('firebase-functions');
 
 const admin = require('firebase-admin');
 admin.initializeApp();
+var db = admin.firestore();
 
-exports.sendNotificationOnDBChange = functions.firestore
-    .document('favors/{favorId}')
-    .onWrite((change, context) => {
+// function to calculate the distance between two points given their coordinates
+function distanceInKm(lon1, lat1, lon2, lat2) {
+    var R = 6371; // Radius of the earth in km
+    var dLat = (lat2 - lat1) * Math.PI / 180;  // Javascript functions in radians
+    var dLon = (lon2 - lon1) * Math.PI / 180;
+    var a = Math.sin(dLat / 2) * Math.sin(dLat / 2) +
+        Math.cos(lat1 * Math.PI / 180) * Math.cos(lat2 * Math.PI / 180) *
+        Math.sin(dLon / 2) * Math.sin(dLon / 2);
+    var c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1 - a));
+    return R * c;
+}
```

Comment:
Maybe we can find libraries that do this for us? Just a thought.

Reply:
Yeah I looked for it but my inexperience with Javascript just made me do this with a manual approach. We should definitely look for some library to handle this in the next sprints. We don't have enough time right now to test another library and check that everything works the same but I'll open an issue for this.

Reply:
Sounds good, thanks!